src/node_process_methods.cc

```diff
@@ -471,6 +473,102 @@ static void ReallyExit(const FunctionCallbackInfo<Value>& args) {
   env->Exit(code);
 }
 
+inline char** copy_js_strings_array(
+  Environment* env, Local<Array> js_array, int* target_length) {
+  Local<Context> context = env->context();
+  char** target = nullptr;
+  int length = js_array->Length();
+
+  CHECK_LT(length, INT_MAX);
+
+  target = new char*[length + 1];
+  target[length] = nullptr;
+
+  if (length > 0) {
+    for (int i = 0; i < length; i++) {
+      node::Utf8Value str(
+        env->isolate(), js_array->Get(context, i).ToLocalChecked());
+      target[i] = strdup(*str);
+      CHECK_NOT_NULL(target[i]);
+    }
+  }
+
+  *target_length = length;
+  return target;
+}
+
+static void Replace(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Local<Context> context = env->context();
+
+  CHECK_GT(args.Length(), 1);
+  CHECK(args[0]->IsString());
+
+  // Prepare arguments and environment
+  char **argv = nullptr;
+  char **envp = nullptr;;
+  int argv_length = 0;
+  int envp_length = 0;
+
+  // Prepare arguments - Note that the pathname is always the first argument
+  if (args.Length() > 1) {
+    CHECK(args[1]->IsArray());
+
+    Local<Array> argv_array = args[1].As<Array>();
+
+    Local<Array> full_argv_array = Array::New(
+      env->isolate(), argv_array->Length() + 1);
+    full_argv_array->Set(context, 0, args[0].As<String>()).Check();
+    for (unsigned int i = 0; i < argv_array->Length(); i++) {
+      full_argv_array->Set(
+        context, i + 1, argv_array->Get(context, i).ToLocalChecked()).Check();
+    }
+
+    argv = copy_js_strings_array(env, full_argv_array, &argv_length);
+  } else {
+    node::Utf8Value pathname_string(env->isolate(), args[0].As<String>());
+
+    argv = new char*[2];
+    argv[0] = strdup(*pathname_string);
+    argv[1] = nullptr;
+  }
+
+  if (args.Length() > 2) {
+    CHECK(args[2]->IsArray());
+    envp = copy_js_strings_array(env, args[2].As<Array>(), &envp_length);
+  }
+
+  // Set stdin, stdout and stderr to be non-close-on-exec
+  // so that the new process will inherit it.
+  for (int i = 0; i < 3; i ++) {
+    int flags = fcntl(i, F_GETFD, 0);
+    if (flags >= 0) {
+      flags &= ~FD_CLOEXEC;
```

Comment:
there is no err check if the `fcntl` function fails ?

Reply:
Good spot. Added.